src/FsAutoComplete/FsAutoComplete.Lsp.fs

```diff
@@ -1188,13 +1190,124 @@ type FSharpLspServer(backgroundServiceEnabled: bool, state: State, lspClient: FS
         let fileName = doc.GetFilePath() |> Utils.normalizePath
         let! res = commands.FormatDocument fileName
         match res with
-        | Ok (lines, formatted) ->
+        | Ok (FormatDocumentResponse.Formatted (lines, formatted)) ->
             let range =
                 let zero = { Line = 0; Character = 0 }
                 let lastPos = lines.GetLastFilePosition()
                 { Start = zero; End = fcsPosToLsp lastPos }
 
             return LspResult.success(Some([| { Range = range; NewText = formatted  } |]))
+        | Ok FormatDocumentResponse.Ignored ->
+          let fileName = UMX.untag fileName |> Path.GetFileName
+          do! lspClient.WindowShowMessage { Type = MessageType.Info; Message = (sprintf "\"%s\" is ignored by a .fantomasignore file." fileName) }
+          return LspResult.success None
+        | Ok FormatDocumentResponse.UnChanged ->
+          return LspResult.success None
+        | Ok FormatDocumentResponse.ToolNotPresent ->
+              let actions =
```

Comment:
Could all this logic be added to Fantomas.Client?  Thanks

Reply:
This is all LSP-specific responses/API usage here, it's not really something that I would expect to drop seamlessly into, say, the VS API.  The bit below about prompting especially would be editor-specific, I have zero idea what those interfaces look like.